scripts/inject-deterministic.py

```diff
@@ -1,17 +1,107 @@
-from __future__ import absolute_import, print_function
+# from __future__ import absolute_import, print_function
 
+import hashlib
 import re
 import time
+
 from mitmproxy import ctx
 
 
 class AddDeterministic():
 
+    def get_csp_directives(self, headers):
+        csp = headers.get("Content-Security-Policy", "")
+        return [d.strip() for d in csp.split(";")]
+
+    def get_csp_script_sources(self, headers):
+        for directive in self.get_csp_directives(headers):
+            if directive.startswith("script-src "):
+                return directive.split()[1:]
+
+    def get_nonce_from_headers(self, headers):
+        """
+        get_nonce_from_headers returns the nonce token from a
+        Content-Security-Policy (CSP) header's script source directive, or an empty
+        string if the CSP header's script source does not contain a nonce.
+        For more background information on CSP and nonce, please refer to
+        https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/
+        Content-Security-Policy/script-src
+        https://developers.google.com/web/fundamentals/security/csp/
+        """
+        for source in self.get_csp_script_sources(headers) or []:
+            if source.startswith("'nonce-"):
+                return source.partition("'nonce-")[-1][:-1]
+        return None
+
+    def get_script_with_nonche(self, script, nonce=None):
+        """
+         Given a nonce, get_script_with_nonche returns the injected script text with the nonce.
+        If nonce is an empty string, getScriptWithNonce returns the script block
+        without attaching a nonce attribute.
+        Some responses may specify a nonce inside their Content-Security-Policy,
+        script-src directive.
+        The script injector needs to set the injected script's nonce attribute to
+        open execute permission for the injected script.
+        """
+        if nonce:
+            return '<script nonce="{}">{}</script>'.format(nonce, script)
+        return '<script>{}</script>'.format(script)
+
+    def get_new_csp_script_diretives(self, headers, sha256):
```

Comment:
I would rename this as it's not returning new directives. It's returning an updated script-src directive.